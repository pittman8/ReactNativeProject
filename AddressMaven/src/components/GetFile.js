import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

const styles = {
    container: {
        flexGrow: 1,
        textAlign: 'center'
    }
};

class GetFile extends Component {
    constructor() {
        super();
        this.state = {
            file: 'unknown'
        };
    }

    render() {
        const {classes} = this.props;
        return (
            <div className={classes.container}>
                <p>File: {this.state.file}</p>
                <Button variant="raised" color="primary" onClick={this.getFile}>
                    Get File
                </Button>
            </div>
        );
    }

    getFile = () => {
        this.setState({
            file: 'url-file.js'
        });
    };
}
export default withStyles(styles)(GetFile);
