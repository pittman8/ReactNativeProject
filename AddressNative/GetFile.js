import React, { Component } from 'react';
import styles from './elf-styles';
import  { Text, View, Button } from 'react-native';

class GetFile extends Component {
    constructor() {
        super();
        this.state = {
            file: 'unknown'
        };
    }

    render() {
        return (
            <View className="GetFile">
                <Text>File: {this.state.file}</Text>
                <View style={styles.buttonView} >
                    <Button
                        onPress={this.getFile}
                        title="Get Data"
                        color="#841584"
                        accessibilityLabel="Learn more about this purple button"
                    />
                </View>
            </View>
        );
    }

    getFile = () => {
        this.setState({
            file: 'url-file.js'
        });
    };
}
export default GetFile;
