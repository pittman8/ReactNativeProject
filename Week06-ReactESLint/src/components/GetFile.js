import React, { Component } from "react";

class GetFile extends Component {
  constructor() {
    super();
    this.state = {
      file: "unknown"
    };
  }

  render() {
    return (
      <div className="GetFile">
        <p className="App-intro">File: {this.state.file}</p>
        <button className="elf" onClick={this.getFile}>
          Get File
        </button>
        <hr />
      </div>
    );
  }

  getFile = () => {
    this.setState({
      file: "url-file.js"
    });
  };
}
export default GetFile;
