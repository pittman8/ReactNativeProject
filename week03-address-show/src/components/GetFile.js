import React, { Component } from 'react';

class GetFile extends Component {
    constructor() {
        super();
        this.state = {
            nine: '0'
        };

    }

    render() {
        return(
            <div className="GetFile">
                <p className="App-intro">
                    Nine: {this.state.nine}
                </p>
                <button className="elf" onClick={this.getNine}>Get Nine</button>
                <hr>
                </hr>
            </div>
        );
    }

    getNine = () => {
        this.setState({
            nine: '9'
        })
    };
}
export default GetFile;