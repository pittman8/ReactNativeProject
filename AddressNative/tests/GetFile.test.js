import React from 'react';
import GetFile from '../GetFile';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Text } from 'react-native';

configure({ adapter: new Adapter() });

describe('GetFile tests', function() {
    it('renders without crashing', () => {
        const rendered = renderer.create(<GetFile />).toJSON();
        expect(rendered).toBeTruthy();
    });

    it('renders default output with react (mount)', () => {
        const tree = renderer.create(<GetFile />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders state of File paragraph before button click', () => {
        const wrapper = shallow(<GetFile />);
        const file = <Text>File: unknown</Text>;
        expect(wrapper.contains(file)).toBe(true);
    });
});
