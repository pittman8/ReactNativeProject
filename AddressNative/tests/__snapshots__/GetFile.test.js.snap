// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetFile tests renders default output with react (mount) 1`] = `
<View
  className="GetFile"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    File: 
    unknown
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#68a0cf",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginLeft": 40,
        "marginRight": 40,
        "marginTop": 10,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "75%",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Get Data
        </Text>
      </View>
    </View>
  </View>
</View>
`;
